tmpauthenticator/__init__.py

```diff
@@ -30,19 +25,20 @@ async def get(self):
         login mechanism. This only happens when /tmplogin is hit - so you can use
         other parts of the hub as you normally would.
         """
-        # Create a new user.
+        username = str(uuid.uuid4())
+
+        # Run post_auth_hook
         #
-        # user_from_username ref: https://github.com/jupyterhub/jupyterhub/blob/4.0.0/jupyterhub/handlers/base.py#L504-L505
+        # run_post_auth_hook ref: https://github.com/jupyterhub/jupyterhub/blob/10e7ab96e5f09db0731e535b22984e2743846fa2/jupyterhub/auth.py#L400-L418
         #
-        username = str(uuid.uuid4())
-        user = self.user_from_username(username)
+        authenticated = {"name": username}
+        authenticated = await self.run_post_auth_hook(self, authenticated)
```

Comment:
Is this always required to be async?

Reply:
Hmmmm, I'm not confident about this.

This is the definition of the function run_post_auth_hook function: https://github.com/jupyterhub/jupyterhub/blob/10e7ab96e5f09db0731e535b22984e2743846fa2/jupyterhub/auth.py#L400-L418

Are you thinking if we should do something like this?


```python
        authenticated = self.run_post_auth_hook(self, authenticated)
        if inspect.isawaitable(authenticated):
            authenticated = await authenticated
```

Reply:
There's no reason for subclasses to override run_post_auth_hook (it should probably have a private name).